Java/core/src/test/java/com/amazon/randomcutforest/CPUTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+import com.amazon.randomcutforest.testutils.ShingledMultiDimDataWithKeys;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.concurrent.ForkJoinPool;
+
+@Tag("functional")
+public class CPUTest {
+
```

Comment:
Can you write the conclusion (async is 10%~15% faster than sync version) and purpose of the tests in the comments? Also, I usually write assertEquals or assertTrue in tests to verify sth, but I cannot find any verification here. Should we add some?

Reply:
Would that conclusion remain true if the IDE/JVM/machine is changed? Also some folks like the JVM to be "warmed", so all those considerations come in before making a statement "X is faster than Y by Z". Also the conclusions can change based on the number of models (in this example 6) being tracked. :) 

The reason for providing this was a tool for folks to "test" the machine/environment they are considering running RCF in parallel. The verification is that the two methods are producing the exact same scores and forecast printed out. 
One could merge them as a test, but then one loses the granularity of time measurement. One could of course write all this up as a benchmarking suite -- but that needs resourcing.

Reply:
Got it. Can you add the purpose of these tests?